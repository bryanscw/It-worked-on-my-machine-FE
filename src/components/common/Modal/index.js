import React, { Component } from 'react';
import $ from 'jquery';

export class Modal extends Component {
    componentDidMount() {
        this.updateModal();
        $(this.modalRef).on('hidden.bs.modal', e => this.props.onClose());
    }
    
    componentDidUpdate(prevProps) {
        if (this.props.isVisible !== prevProps.isVisible) {
            this.updateModal();
        }
    }

    updateModal() {
        if (this.props.isVisible)
            $(this.modalRef).modal('show');
        else {
            $(this.modalRef).modal('hide');
        }
    }

    render() {
        const {
            title,
            children,
            onClose,
        } = this.props;

        return (
            <div className="modal fade" ref={modalRef => this.modalRef = modalRef} tabIndex="-1" data-testid="modal">
                <div className="modal-dialog">
                    <div className="modal-content">
                        <div className="modal-header">
                            <h5 className="modal-title">{title}</h5>
                            <button type="button" className="close" onClick={onClose}>
                                <span aria-hidden="true">&times;</span>
                            </button>
                        </div>
                        <div className="modal-body">
                            {children}
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default Modal
