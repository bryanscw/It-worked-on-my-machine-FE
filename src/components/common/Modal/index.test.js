import React from 'react'
import { render, cleanup, fireEvent, waitForDomChange } from '@testing-library/react'
import Modal from './index.js';

afterEach(cleanup);

it('should take a snapshot', () => {
    const { asFragment } = render(
        <Modal
            title="Modal title"
            onClose={() => {}}
            isVisible={true}>
            <p>Modal body</p>
        </Modal>
    );

    expect(asFragment()).toMatchSnapshot();
})

it('should hide modal the close button is clicked', async () => {
    const { getByText, getByTestId } = render(
        <Modal
            title="Modal title"
            onClose={() => {}}
            isVisible={true}>
            <p>Modal body</p>
        </Modal>
    );

    const modal = getByTestId("modal");
    expect(modal).toBeVisible();

    fireEvent.click(getByText('Ã—'));
    await waitForDomChange(() => expect(modal).not.toBeVisible());
})