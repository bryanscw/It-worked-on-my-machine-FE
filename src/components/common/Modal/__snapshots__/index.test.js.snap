// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal fade"
    data-testid="modal"
    style="padding-right: 0px;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
          >
            Modal title
          </h5>
          <button
            class="close"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          class="modal-body"
        >
          <p>
            Modal body
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
